src/CIQS-CLI-Extension/azext_ciqs/custom.py

```diff
@@ -30,15 +54,42 @@ def deployDeployment(cmd, deploymentId, subscription=None):
     print('To be implemented')
 
 def viewDeployment(cmd, deploymentId, subscription=None):
-    print('To be implemented')
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    if subscription is None:
+        subscription = get_subscription_id(cmd.cli_ctx)
+    profile = Profile(cli_ctx=cmd.cli_ctx)
+    auth_token = profile.get_raw_token(subscription=subscription)
+    path = '/api/deployments/' + subscription + '/' + deploymentId
+    return _makeAPICall(cmd, 'GET', path, auth_token=auth_token)
 
 def deleteDeployment(cmd, deploymentId, subscription=None):
-    print('To be implemented')
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    if subscription is None:
+        subscription = get_subscription_id(cmd.cli_ctx)
+    profile = Profile(cli_ctx=cmd.cli_ctx)
+    auth_token = profile.get_raw_token(subscription=subscription)
+    path = '/api/deployments/' + subscription + '/' + deploymentId
+    return _makeAPICall(cmd, 'DELETE', path, auth_token=auth_token)
 
-# group ciqs gallery
+#---------------------------------------------------------------------------------------------
+# sub-group ciqs gallery
+#---------------------------------------------------------------------------------------------
 
 def listTemplates(cmd):
-    print('To be implemented')
+    path = '/api/gallery'
+    return _makeAPICall(cmd, 'GET', path)
 
 def getTemplate(cmd, templateId):
-    print('To be implemented')
\ No newline at end of file
+    profile = Profile(cli_ctx=cmd.cli_ctx)
+    auth_token = profile.get_raw_token()
+    path = '/api/gallery/' + templateId
+    return _makeAPICall(cmd, 'GET', path, auth_token=auth_token)
+
+def locations(cmd, templateId, subscription=None):
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    if subscription is None:
+        subscription = get_subscription_id(cmd.cli_ctx)
+    profile = Profile(cli_ctx=cmd.cli_ctx)
+    auth_token = profile.get_raw_token(subscription=subscription)
+    path = '/api/locations/' + templateId + '/' + subscription
+    return _makeAPICall(cmd, 'GET', path, auth_token=auth_token)
```

Comment:
Same suggestion with listTemplates API. Moreover, consider adding error handling and recovery message in the output, such that:

```
def tranformResponse(locationResponse):
    if (len(locationResponse) == 0):
        throw Error(HttpCode.NotFound, "No location is found in subscription 123 for solution ‘’. Please use another subscription.")
    else:
        return tranformLocationResponse(locationResponse)
```